opencog/atoms/pattern/PatternLink.cc

```diff
@@ -23,6 +23,8 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#include <boost/range/algorithm/find_if.hpp>
```

Comment:
I agree with boost in general, but the range library is fairly simple and stable and saves a bit of the verbosity of the STL algorithms.